src/settings/settinggroup.cpp

```diff
@@ -0,0 +1,87 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "settings/settinggroup.h"
+
+#include "leakdetector.h"
+#include "settingfactory.h"
+#include "settingsmanager.h"
+
+SettingGroup::SettingGroup(const QString& groupKey, bool removeWhenReset,
+                           bool sensitiveSetting, QStringList acceptedKeys)
+    : m_groupKey(groupKey),
+      m_sensitiveSetting(sensitiveSetting),
+      m_removeWhenReset(removeWhenReset),
+      m_acceptedKeys(acceptedKeys) {
+  MZ_COUNT_CTOR(Setting);
+
+  // Group settings are dynamic, therefore we need to load from memory all
+  // settings that exist under this group prefix in order to emit change signals
+  // when they change.
+  auto sb = SettingsManager::instance();
+  sb->m_settings.beginGroup(m_groupKey);
+  QStringList keys = sb->m_settings.allKeys();
+  sb->m_settings.endGroup();
```

Comment:
Consider replacing this block with SettingManager::instance()->getKeysInGroup(m_groupKey), which returns the list of keys of Settings  in the group.

Reply:
I have addressed this and added it to the PRE-WORK PR. I did not do exactly what you propose here. I'd not like to add public methods to change the underlying settings to the SettingsManager, that is an invitation for mess ups. Instead, I created a connector for the QSettings that is passed to the Setting and SettingGroup for them to be able to access the QSettings. Let me know what you think about the approach.